stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -571,6 +571,14 @@ end
 
     @test tr(D) == 10
     @test det(D) == 4
+
+    # sparse matrix block diagonals
+    s = SparseArrays.sparse([1 2; 3 4])
+    D = Diagonal([s, s])
+    @test isa(D, Diagonal{SparseMatrixCSC{Int64,Int64},Vector{SparseMatrixCSC{Int64,Int64}}})
```

Comment:
This should have been just
```julia
@test isa(D, Diagonal{SparseMatrixCSC{Int64,Int}})
```
I guess.

Reply:
Or even less specific, to avoid 32- vs 64-bit issues
```suggestion
    @test isa(D, Diagonal{SparseMatrixCSC})
```
That should be enough to test, because `SparseMatrixCSC` is not a subtype of `Matrix`.